### DISCUSSION TITLE
Binding with @Bindable var store: Store

### DISCUSSION THREAD

**Discussion by BaochengHan-TomTom** [pointfreeco/swift-composable-architecture#2783] [Q&A]
Hi there,
According to the the migration guide for 1.7, the way how we use `binding` changed like [here](https://github.com/pointfreeco/swift-composable-architecture/blob/observation-beta/Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.7.md#viewstorebinding)

what about if my logic is like this before upgrade
```swift
TabView(
  selection: viewStore.binding(get: \.tab, send: { .tabChanged($0 * 2) })
) {
  // ...
}
```
how to adapt to the new API then?
Apparently it doesn't work like following as the parameter of `sending` needs to be a `CaseKeyPath` 
```swift
TabView(
  selection: $store.tab.sending(get: \.tabChanged($0 * 2))
) {
  // ...
}
```

**Comment by Matejkob** [comment]
Why not move this logic to the reducer? 

```swift
var body: some ReducerOf<Self> {
// ...
case let .tabChanged(newTab):
  let modifiedTab = newTab * 2
  // Do something with `modifiedTab`
  return .none
// ...
}
```

**Reply by mbrandonw** [reply]
Yeah I agree. Better to put that logic in the reducer so that you can test it, and best to keep sending actions from the view as simple and logicless as possible.
